[Adi Nata] iP

adinata15 opened the pull request:
Refactor program to follow SLAP standard

adinata15 committed:
Added greeting on Duke

adinata15 committed:
Added Duke's greeting

adinata15 committed:
Add tag and fix comment

adinata15 committed:
Implement Greet, Echo, Exit

adinata15 committed:
Add Level 2. Add, List
Add the ability to store whatever text entered by the user and display
them back to the user when requested.

adinata15 committed:
Level 3. Mark as Done
Added mark as done,modify displayList method and new Task class.

adinata15 committed:
A-CodingStandard: Follow the Coding Standard
Tweak and refactor the code to comply with a given coding standard.

adinata15 committed:
Level 4. ToDos, Events, Deadlines
Add support for tracking three types of tasks:
*ToDos: tasks without any date/time attached to it
e.g., visit new theme park.
*Deadlines: tasks that need to be done before a specific date/time
e.g., submit report by 11/10/2019 5pm.
*Events: tasks that start at a specific time and ends at a specific time
e.g., team project meeting on 2/10/2019 2-4pm.

adinata15 committed:
A-CodeQuality. Improve code quality
Critically adjust and refactor code to follow SLAP and good coding
standards.

adinata15 committed:
A-TextUiTesting. Automated Text UI Testing
Use the input/output redirection technique to semi-automate the testing
of Duke.

poonchuanan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,171 @@
+
 public class Duke {
+    public static final int MAX_LIST_SIZE = 100;
+    private static Task[] tasks=new Task[MAX_LIST_SIZE];//user's to-do-list (max 100 items)
```
Hi, operators should be surrounded by a space character, just like line 6.
Good use of named constant 'MAX_LIST_SIZE'!

imhm reviewed (commented):
Good code quality!

johanesrafael commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,171 @@
+    private static void markDone(String userInput) {
+        //take word after "done" as task index
+        String taskIndexString = userInput.split(" ")[1];
+        Integer taskIndex = Integer.parseInt(taskIndexString);
```
Hello! Perhaps, you can add comment on this one for signifying a conversion (for example: // convert taskIndexString into an integer)

johanesrafael reviewed (commented):
Looks good! overall, nice coding

johanesrafael commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,58 @@
+    public void displayListPerTask(int taskIndex){
+        System.out.printf(" %d.[%c][%s] %s\n"
+                ,taskIndex+1,getTaskType(),getStatusIcon(),getDescription()
+        );
```
Hello! you may pay attention to above. the ); is better in line 55 or you can use what you've done before in printing message(Egyptian style)

johanesrafael commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,37 @@
+    public void displayListPerTask(int taskIndex){
+        System.out.printf(" %d.[%c][%s] %s (at: %s)\n"
+                ,taskIndex+1,getTaskType(),getStatusIcon(),getDescription(),at
+        );
```
Hello! you may pay attention to above. the ); is better in line 33 or you can use what you've done before in printing message(Egyptian style)

johanesrafael reviewed (commented):
Overall looks good! don't need any big change.

johanesrafael commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,171 @@
+        String userInput; //user input
+        
+        //scan user's input
+        userInput = input.nextLine();
```
Hello again! perhaps, you can make this in line 115 such that it is not redundant like "String userInput = input.nextLine(); and also can comment it as //scan user input

johanesrafael reviewed (commented):
Just an additional comment

adinata15 committed:
Add suggestion from Peer Review

adinata15 committed:
Level 5. Handle Errors
Teach Duke to deal with errors such as incorrect inputs entered by the
user.

adinata15 committed:
Merge branch 'branch-Level-5' into master

adinata15 committed:
A-Exceptions: Add DukeException

Add EmptyTodoException and InvalidCommandException.

adinata15 committed:
Merge branch 'branch-A-Exceptions' into master

adinata15 committed:
A-Packages. Divide classes into packages
Organise files into packages

adinata15 committed:
A-Packagas. Edit import to single file format
Edit package import statement to single file format.

adinata15 committed:
Merge branch 'branch-A-Packages' into master

adinata15 committed:
Level 6. Delete
Add support for deleting tasks from the list.

adinata15 committed:
A-Collections. Use Java Collections classes
Use Java Collections classes for storing data. For example, you can use
an ArrayList<Task> to store the tasks.

adinata15 committed:
Level 7. Save
Save the tasks in the hard disk automatically whenever the task list
changes. Load the data from the hard disk when Duke starts up.

adinata15 committed:
Merge branch 'branch-Level-6' into master

adinata15 committed:
Merge branch 'branch-Level-7' into master

adinata15 committed:
A-Jar. Package the App as a JAR file
Create JAR package and refactor code to coding standard.

Weng-Kexin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,356 @@
+    }
+
+    //display file not found error message
+    private static void printFileNotFound() {
+        System.out.println(" Are you first time user?\n" +
+                " I am unable to find your past data\n" +
+                "____________________________________________________________");
+    }
```
Consider moving all your print functionalities to a separate class dedicated for printing out messages

Weng-Kexin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,356 @@
+    }
+
+    //add deadline from file
+    private static void addDeadlineFile(String[] taskDetails) {
+        Deadline deadline = new Deadline(taskDetails[2], TaskType.DEADLINE, taskDetails[3]);
+        tasks.add(deadline);
+        if (taskDetails[1].equals("1")) {
+            tasks.get(tasks.size() - 1).markAsDone();
+        }
+    }
+
+    //add event from file
+    private static void addEventFile(String[] taskDetails) {
+        Event event = new Event(taskDetails[2], TaskType.EVENT, taskDetails[3]);
+        tasks.add(event);
+        if (taskDetails[1].equals("1")) {
+            tasks.get(tasks.size() - 1).markAsDone();
+        }
+    }
+
+    //add todo from file
+    private static void addToDoFile(String[] taskDetails) {
+        ToDo toDo = new ToDo(taskDetails[2], TaskType.TODO);
+        tasks.add(toDo);
+        if (taskDetails[1].equals("1")) {
+            tasks.get(tasks.size() - 1).markAsDone();
+        }
+    }
+
```
All 3 functions here are doing the same thing, can consider refactoring into one class so that you don't have to repeat what you wrote 3x

Weng-Kexin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,356 @@
+            } catch (EmptyTodoException e) {
+                displayEmptyTodoMessage();
+            }
+            break;
+        case "deadline":
+            addDeadline(userInput);
+            break;
+        case "event":
+            addEvent(userInput);
+            break;
```
What are the break lines used for here? What happens if it is removed?

Weng-Kexin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,356 @@
+        TaskType taskType;
+        String description;
+
+        //start and end index for description
+        int taskDescriptionStartIndex;
+        int taskDescriptionEndIndex;
+
+        //find start of task description
+        taskDescriptionStartIndex = userInput.indexOf(" ");
+        taskDescriptionEndIndex = userInput.indexOf("/at");
+        //assign task's information
+        description = userInput.substring(taskDescriptionStartIndex + 1, taskDescriptionEndIndex).trim();
+        String at = userInput.substring(taskDescriptionEndIndex + 3).trim();//+3 to exclude "/at"
+        taskType = TaskType.EVENT;
+        //create and assign information to new event
+        Event event = new Event(description, taskType, at);
+        tasks.add(event);
```
Excessive comments (comments are not required when the code is self explanatory)

Weng-Kexin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,356 @@
+        TaskType taskType;
+        String description;
+
+        //start and end index for description
+        int taskDescriptionStartIndex;
+        int taskDescriptionEndIndex;
+
+        //find start of task description
+        taskDescriptionStartIndex = userInput.indexOf(" ");
+        taskDescriptionEndIndex = userInput.indexOf("/by");
+        //assign task's information
+        description = userInput.substring(taskDescriptionStartIndex + 1, taskDescriptionEndIndex).trim();
+        String by = userInput.substring(taskDescriptionEndIndex + 3).trim();//+3 to exclude "/by"
+        taskType = TaskType.DEADLINE;
+        //create and assign information to new deadline
```
Same goes for these comments

Weng-Kexin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,356 @@
+        TaskType taskType;
+        String description;
+
+        //start index for description
+        int taskDescriptionStartIndex;
+
+        //find start of task description
+        taskDescriptionStartIndex = userInput.indexOf(" ");
+        //assign task's information
+        description = userInput.substring(taskDescriptionStartIndex + 1);
+        //check whether todo description is valid
+        if (taskDescriptionStartIndex != 4 || description.isEmpty()) { //4 is for "todo" letter
+            throw new EmptyTodoException();
+        }
+        taskType = TaskType.TODO;
+        //create and assign information to new ToDo
+        ToDo toDo = new ToDo(description, taskType);
```
Here too

adinata15 committed:
A-MoreOOP. Make code more OOP
Refactor the code to extract out closely related code as classes.

adinata15 committed:
Level 8. Dates and Times
Enable Duke to understand and manipulate date format.

adinata15 committed:
Level 9. Find
Give user a way to find task using specified keywords.

adinata15 committed:
A-JavaDoc
Add JavaDOC comments.

adinata15 committed:
Merge pull request #1 from adinata15/branch-Level-8

Level 8. Dates and Times

adinata15 committed:
Merge branch 'master' into branch-Level-9

adinata15 committed:
Remove user data file

adinata15 committed:
Merge pull request #2 from adinata15/branch-Level-9

Add find functionality.

adinata15 committed:
Merge branch 'master' into branch-A-JavaDoc

adinata15 committed:
Merge pull request #3 from adinata15/branch-A-JavaDoc

Add JavaDoc comments.

adinata15 committed:
Update README.md

Add README for Duke

adinata15 committed:
Update README.md

Remove separating lines

adinata15 committed:
Update README

adinata15 committed:
Update README based on Duke output

adinata15 committed:
Fix unsaved delete bug
Fix uncaptured deleted data from "delete" command.

adinata15 committed:
Fix unsaved done bug
- Fix unsaved done task bug.
- Update README to refer to suitable jar file name.

adinata15 committed:
Minor edit to jar name
